legal-api/src/legal_api/core/filing.py

```diff
@@ -61,17 +62,44 @@ def id(self) -> str:  # pylint: disable=invalid-name; defining the std ID
                 self._id = self._storage.id
         return self._id
 
+    @property
+    def filing_type(self) -> str:
+        """Property containing the filing type."""
+        if not self._filing_type and self._storage:
+            self._filing_type = self._storage.filing_type
+        return self._filing_type
+
     @property
     def raw(self) -> Optional[Dict]:
         """Return the raw, submitted and unprocessed version on the filing."""
         if not self._raw:
-            return {}
+            if self._storage:
+                self._raw = self._storage.json
+            else:
+                return {}
         return self._raw
 
     @property
     def json(self) -> Optional[Dict]:
         """Return a dict representing the filing json."""
-        return self._raw
+        _json = {}
+        if self._storage:
+            if self._storage.status == Filing.Status.COMPLETED.value:
+                _json = VersionedBusinessDetailsService. \
+                    get_revision(self.filing_type, self.id, self._storage.business_id)
+            else:
+                _json = self.raw
+        return _json
+
+    @property
+    def storage(self) -> Optional[FilingStorage]:
```

Comment:
Not sure whether this storage which is the filing model itself should be exposed as a property like this or you should have getters for each of the property defined in the property list.

Reply:
I had to expose model for the purpose of get_pdf

Reply:
@lekshmimallika-aot I suggested we can move this in in stages as needed, and up to @vysakh-menon-aot to suggest in the PR where the lines should be.
Can this property be marked as Deprecated with a note, in case someone else does the report updates?

Reply:
Sure. Sounds good